x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/ExpressionCoreWritables.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.core.expression;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExpressionCoreWritables {
```

Comment:
Indeed the reason for that was that each implementation subclass gained a `getNamedWritables()` so one (me :) ) that returned not the class writable but the entire hierarchy writables.
Which is confusing since Field.getNamedWritable() not only it's possible but doesn't return FieldAttribute.ENTRY but also MetadataAttribute and others. Plus it added a cyclic dependency.

I ended up working on it since the CsvTests had errors due to LookupJoinExec serialization not being written in and the serialization code style was split - see the change in EsqlPlugin.getNamedWritables (mix of utility classes plus individual entries).
I'll work look into extracting it in a separate PR.